src/main/java/de/funfried/netbeans/plugins/external/formatter/ui/editor/ExternalFormatterReformatTaskFactory.java

```diff
@@ -9,37 +9,37 @@
  */
 package de.funfried.netbeans.plugins.external.formatter.ui.editor;
 
+import de.funfried.netbeans.plugins.external.formatter.FormatterServiceDelegate;
+import de.funfried.netbeans.plugins.external.formatter.MimeType;
+import de.funfried.netbeans.plugins.external.formatter.ui.Icons;
+import de.funfried.netbeans.plugins.external.formatter.ui.options.Settings;
 import java.lang.ref.Reference;
 import java.lang.ref.WeakReference;
 import java.util.Collection;
 import java.util.Map;
 import java.util.WeakHashMap;
 import java.util.prefs.Preferences;
-
 import javax.swing.SwingUtilities;
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Document;
-
 import org.netbeans.api.annotations.common.NonNull;
 import org.netbeans.api.editor.mimelookup.MimeLookup;
 import org.netbeans.api.editor.mimelookup.MimePath;
+import org.netbeans.api.editor.mimelookup.MimeRegistration;
 import org.netbeans.modules.editor.indent.spi.Context;
 import org.netbeans.modules.editor.indent.spi.ExtraLock;
 import org.netbeans.modules.editor.indent.spi.ReformatTask;
 import org.openide.awt.NotificationDisplayer;
 import org.openide.awt.StatusDisplayer;
 
-import de.funfried.netbeans.plugins.external.formatter.FormatterServiceDelegate;
-import de.funfried.netbeans.plugins.external.formatter.MimeType;
-import de.funfried.netbeans.plugins.external.formatter.ui.Icons;
-import de.funfried.netbeans.plugins.external.formatter.ui.options.Settings;
-
 /**
  * Custom implementation of the {@link ReformatTask.Factory} which delegates the formatting
  * tasks to the configured external formatter or to the internal NetBeans formatter.
  *
  * @author bahlef
  */
+
+@MimeRegistration(mimeType="text/x-sql",service=ReformatTask.Factory.class)
```

Comment:
Is this really necessary? I think it shouldn't, as this is already covered by the entries of the layers.xml.
If it is not necessary, please remove it. If it is necessary, please add a comment why.

Reply:
Necessary, because the Formatter would not engage for SQL files without it. Took me longer to figure that out than crafting the whole PR.
The entries in layers.xml alone did not work. Please do not ask me, why. I am just an accountant.

Reply:
Hmm, I really don't like this one, as it is a specific annotation for SQL formatting in a general class for all formatters. But I'll check if there is any way to get around this and if not, it's fine of course 👍

Reply:
> as it is a specific annotation for SQL formatting in a general class for all formatters. 

Actually this has been my concern too. Turns out that it did not harm anything. Only SQL is formatted by JSQLFormatter, JAVA still goes into the Java Formatters and so on.

I would not mind if you find any better solution, but as far as I have been able to lookup the documentation on that is weak. My theory is, that SQL has been left out from Netbeans formatting since (in the meaning that not only there was no formatter, but there was not even a registration for the event for SQL mimetype).